ui/app/serializers/clients/activity.js

```diff
@@ -2,67 +2,63 @@ import ApplicationSerializer from '../application';
       if (ns['namespace_id'] === 'root') ns['namespace_path'] = 'root';
       let label = ns['namespace_path'];
-      let flattenedNs = {};
-      // we don't want client counts nested within the 'counts' object for stacked charts
-      Object.keys(ns['counts']).forEach((key) => (flattenedNs[key] = ns['counts'][key]));
-      flattenedNs = this.homogenizeClientNaming(flattenedNs);
-
+      let flattenedNs = this.flattenDataset(ns);
       // if no mounts, mounts will be an empty array
-      flattenedNs.mounts = ns.mounts
-        ? ns.mounts.map((mount) => {
-            let flattenedMount = {};
-            let label = mount['mount_path'];
-            Object.keys(mount['counts']).forEach((key) => (flattenedMount[key] = mount['counts'][key]));
-            flattenedMount = this.homogenizeClientNaming(flattenedMount);
-            return {
-              label,
-              ...flattenedMount,
-            };
-          })
-        : [];
-
+      flattenedNs.mounts = [];
+      if (ns?.mounts && ns.mounts.length > 0) {
+        flattenedNs.mounts = ns.mounts.map((mount) => {
+          return {
+            label: mount['mount_path'],
+            ...this.flattenDataset(mount),
+          };
+        });
+      }
       return {
         label,
         ...flattenedNs,
       };
     });
   }
 
-  flattenByMonths(payload, isNewClients = false) {
-    const sortedPayload = [...payload];
+  formatByMonths(monthsArray) {
+    const sortedPayload = [...monthsArray];
+    // months are always returned from the API: [mostRecent...oldestMonth]
     sortedPayload.reverse();
-    if (isNewClients) {
-      return sortedPayload?.map((m) => {
-        return {
-          month: parseAPITimestamp(m.timestamp, 'M/yy'),
-          entity_clients: m.new_clients.counts.entity_clients,
-          non_entity_clients: m.new_clients.counts.non_entity_clients,
-          clients: m.new_clients.counts.clients,
-          namespaces: this.flattenDataset(m.new_clients.namespaces),
-        };
-      });
-    } else {
-      return sortedPayload?.map((m) => {
+    return sortedPayload.map((m) => {
+      if (Object.keys(m).includes('counts')) {
+        let totalClients = this.flattenDataset(m);
+        let newClients = this.flattenDataset(m.new_clients);
         return {
           month: parseAPITimestamp(m.timestamp, 'M/yy'),
-          entity_clients: m.counts.entity_clients,
-          non_entity_clients: m.counts.non_entity_clients,
-          clients: m.counts.clients,
-          namespaces: this.flattenDataset(m.namespaces),
+          ...totalClients,
+          namespaces: this.formatByNamespace(m.namespaces),
           new_clients: {
-            entity_clients: m.new_clients.counts.entity_clients,
-            non_entity_clients: m.new_clients.counts.non_entity_clients,
-            clients: m.new_clients.counts.clients,
-            namespaces: this.flattenDataset(m.new_clients.namespaces),
+            month: parseAPITimestamp(m.timestamp, 'M/yy'),
```

Comment:
it may behoove us to eventually refactor and keep this a timestamp, and instead in the charts use d3's `scaleTime` (instead of `scalePoint` or `scaleLinear`), and format to `M/yy` there instead. This would prevent awkward date manipulation functions (like `formatChartDate`)